### ISSUE TITLE
Different behavior to mypy when using local stubs 

### ISSUE THREAD

**Issue by KotlinIsland** [pre-commit/mirrors-mypy#20]
```
foo
├── __init__.py
├── a.py
├── b.py
└── b.pyi
```
.pre-commit-config.yaml
```yaml
repos:
  - repo: https://github.com/pre-commit/mirrors-mypy
    rev: v0.770
    hooks:
    - id: mypy
      args: ["--strict"]
```
a.py:
```py
from .b import b
reveal_type(b)
```
b.py:
```py
```
b.pyi
```
b: int
```
```
> pre-commit run mypy -a
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

foo\a.py:1: error: Module 'foo.b' has no attribute 'b'
foo\a.py:3: note: Revealed type is 'Any'
Found 1 error in 1 file (checked 3 source files)

> mypy --strict foo
foo\a.py:3: note: Revealed type is 'builtins.int'
```

**Comment by asottile** [comment]
I feel a bit like a broken record :)

[pre-commit passes filenames as positional arguments](https://pre-commit.com/#passing-arguments-to-hooks)

**Comment by KotlinIsland** [comment]
```
> mypy --strict foo/__init__.py
Success: no issues found in 1 source file
> mypy --strict foo/a.py
foo\a.py:3: note: Revealed type is 'builtins.int'
> mypy --strict foo/b.py
Success: no issues found in 1 source file
> mypy --strict foo/b.pyi
Success: no issues found in 1 source file
```
Okay, but this passes fine when I use mypy.

**Comment by asottile** [comment]
please read the docs: `mypy --strict foo/__init__.py foo/a.py foo/b.py`

**Comment by KotlinIsland** [comment]
why not include b.pyi?
```
>mypy --strict foo/__init__.py foo/a.py foo/b.pyi
foo\a.py:3: note: Revealed type is 'builtins.int'
```

**Comment by asottile** [comment]
pyi is not matched by [`types: python`](https://github.com/pre-commit/mirrors-mypy/blob/323f177eb169987f7034a75d0a195e4448eec0ba/.pre-commit-hooks.yaml#L5)

**Comment by KotlinIsland** [comment]
That's a pretty big departure from pure mypy, do you think the hook could be updated to match py and pyi?

**Comment by KotlinIsland** [comment]
But then mypy would complain about duplicate modules
```
> mypy --strict foo/__init__.py foo/a.py foo/b.py foo/b.pyi
foo\b.pyi: error: Duplicate module named 'foo.b' (also at 'foo/b.py')
Found 1 error in 1 file (checked 4 source files)
```

**Comment by asottile** [comment]
idk man, play with the [documented configuration options](https://pre-commit.com/#pre-commit-configyaml---hooks) and try some stuff out, pre-commit is literally just forwarding files along to mypy and the rest is whatever mypy decides to do with it
